### PR TITLE
pac-cmd changes for issue 5354 and 5346

### PR DESCRIPTION
**Pull request by joesis**
This PR has two changes:

1. Match URL prefix when turning off PAC, for https://github.com/getlantern/lantern/issues/5354
2. Added `show` option to verify if the PAC is set or not, to help to troubleshoot https://github.com/getlantern/lantern/issues/5346

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,47 +4,60 @@
...
+    else
+        ifeq ($(PROCESSOR_ARCHITECTURE),AMD64)
+            CCFLAGS += -D AMD64
+            BIN = binaries/windows/pac_amd64
+        endif
+        ifeq ($(PROCESSOR_ARCHITECTURE),x86)
+            CCFLAGS += -D IA32
+            BIN = binaries/windows/pac_386
+        endif
+    endif
```
Comment (joesis) [inline Makefile]: The whole bunch of changes above is to generate different binaries on 32bit and 64bit Windows. See https://blogs.msdn.microsoft.com/david.wang/2006/03/27/howto-detect-process-bitness/ for reference.

File: Makefile [hunk 2]
```diff
@@ -4,47 +4,60 @@
...
+    LDFLAGS += -l rasapi32 -l wininet -Wl,--subsystem,windows
 else
-	UNAME_S := $(shell uname -s)
```
Comment (joesis) [inline Makefile]: All changes below are simply replacing tabs with spaces.

**Comment by joesis** [comment]
@oxtoacart could you have a look as you recently touched similar code? The .c files are changed significantly to add the `show` option without repetitive code, but the logic remains the same.

File: README.md [hunk 3]
```diff
@@ -9,11 +9,12 @@ Note - you will need to run make separately on each platform.
 # Usage
 
 ```sh
+pac show
```
Comment (oxtoacart) [inline README.md]: Maybe also explain what pac show does?

File: main.c [hunk 4]
```diff
@@ -20,7 +20,9 @@ int main(int argc, char* argv[]) {
   }
 #endif
 
-  if (strcmp(argv[1], "on") == 0) {
+  if (strcmp(argv[1], "show") == 0) {
+    show();
```
Comment (oxtoacart) [inline main.c]: Why do we ignore the return value from show()

**Comment by oxtoacart** [comment]
@joesis I went ahead and signed the darwin binary.

File: main.c (hunk 4 above)
Comment (oxtoacart) [inline main.c]: Note also that the darwin version of show() actually returns TRUE, which is an error status.

**Comment by joesis** [comment]
@oxtoacart I updated `main.c` per your comment. Do you have other comments on the code? Needs to rebuild on all platforms.

File: darwin.c [hunk 5]
```diff
@@ -2,37 +2,55 @@
...
+  }
+
+  success = SCNetworkProtocolSetConfiguration(proxyProtocolRef, (__bridge CFDictionaryRef)newPreferences);
+  if(!success) {
+    NSLog(@"Failed to set Protocol Configuration");
+  }
+  return success;
+}
+
+int visit(visitor v, bool persist, bool turnOn, const char* pacUrl)
```
Comment (oxtoacart) [inline darwin.c]: v is technically not a "visitor" in the traditional sense of the word, but it's not a big deal.

File: darwin.c [hunk 6]
```diff
@@ -2,37 +2,55 @@
...
+      [newPreferences setValue:[NSNumber numberWithInt:0] forKey:(NSString*)kSCPropNetProxiesProxyAutoConfigEnable];
+      [newPreferences setValue:@"" forKey:(NSString*)kSCPropNetProxiesProxyAutoConfigURLString];
+    }
+  }
+
+  success = SCNetworkProtocolSetConfiguration(proxyProtocolRef, (__bridge CFDictionaryRef)newPreferences);
+  if(!success) {
+    NSLog(@"Failed to set Protocol Configuration");
+  }
+  return success;
```
Comment (oxtoacart) [inline darwin.c]: Oh, I understand now. This boolean gets translated into an error before returning from `show()` so it's okay.

**Comment by oxtoacart** [comment]
LGTM!

**Comment by oxtoacart** [comment]
Tests pass on Darwin and Windows XP (32 bit) and 10 (64 bit).

Nice work!

File: darwin.c (hunk 5 above)
Comment (joesis) [inline darwin.c]: Sorry I picked this name by instinct rather than carefully thought.

File: darwin.c (hunk 6 above)
Comment (joesis) [inline darwin.c]: I admit it's obscure

**Comment by joesis** [comment]
Thanks for reviewing and testing @oxtoacart

File: darwin.c (hunk 5 above)
Comment (oxtoacart) [inline darwin.c]: It does feel almost like a visitor :)  It's no big deal.

File: darwin.c (hunk 6 above)
Comment (oxtoacart) [inline darwin.c]: It's not that obscure, I just didn't read carefully enough the first time.

**Pull request merged**